client/src/js/services/receipts/ReceiptModal.js

```diff
@@ -43,31 +43,34 @@ function ReceiptModal(Modal, Receipts) {
    */
   function invoice(uuid, notifyCreated) {
 
-    /** @todo Discuss if these should be overridable from the controller or if the config should be set here */
-    const options = {
-      title         : 'PATIENT_INVOICE.PAGE_TITLE',
-      createdKey    : 'PATIENT_INVOICE.SUCCESS',
-      identifier    : 'reference',
-      renderer      : Receipts.renderer,
-      notifyCreated,
-    };
-
-    const receiptOptions = {
-      renderer      : Receipts.renderer,
-      currency : Receipts.receiptCurrency,
-    };
-
-    const invoiceRequest = Receipts.invoice(uuid, receiptOptions);
-    const invoiceProvider = {
-      resolve : {
-        receipt : function receiptProvider() { return { promise : invoiceRequest }; },
-        options : function optionsProvider() { return options; },
-      },
-    };
-
-    const configuration = angular.extend(modalConfiguration, invoiceProvider);
-    const instance = Modal.open(configuration);
-    return instance.result;
+    return Receipts.invoicePosted(uuid).then(isPosted => {
```

Comment:
If we add a `posted` column to the invoice table, we can use the `InvoiceService.read()` method to get both the `posted` and `reversed` values for this invoice.

We should also put this into the `invoiceProvider` instead of firing it before the modal is opened.  It will improve the application's performance.
```js
const invoiceProvider = {
  resolve : {
    receipt : () => ({ promise : invoiceRequest }),
    options : () => options,
    invoice : () => Invoice.read(uuid),
  }
};
```

That way, the modal can open, then we can just check the `data.reversed` and `data.posted` values to figure out if we should display the `posted` and `reversed` signs.